core/src/services/providers/snyk.rs

```diff
@@ -0,0 +1,194 @@
+                        Ok(option_sbom) => {
+                            match option_sbom {
+                                Some(sbom) =>{
+                                    valid_sboms.push((sbom, project.clone()));
+                                    println!("Adding valid SBOM to list for project: {}, from org: {}", project.id, org_projects.org.id.clone().unwrap());
+                                    
+                                },
+                                None => println!("No SBOM found for project: {}, from org: {}", project.id, org_projects.org.id.clone().unwrap()),
+                            }
+                        },
+                        Err(err) => println!("{}", err),
+
+                    }
+                },
+                _ => continue 
+            }
+        }
+       // return valid_sboms;
+    }
+    
+    async fn get_valid_sboms(&self, snyk_token: String) -> Vec<(HashMap<String, Value>, ProjectDetails)>{
+
+        let start = Instant::now();
+        let mut valid_sboms:  Vec<(HashMap<String, Value>, ProjectDetails)> = Vec::new();
+
+        println!("Starting SnykProvider scan...");
+        
+        println!("Scanning Snyk for SBOMS...");
+
+        let snyk_data = self.snyk_api.get_orgs(snyk_token.clone()).await;
+
+        match snyk_data {
+            Ok(data) => {
+                let get_projects_tuple_result = self.snyk_api.get_projects_from_org_list(snyk_token.clone(), data).await;
+
+                for errors in get_projects_tuple_result.1 {
+                    println!("{}", errors)
+                }
+
+                for orgs in get_projects_tuple_result.0.iter() {
+                    self.get_all_snyk_sboms(snyk_token.clone(), orgs.clone(), &mut valid_sboms).await;
+                }
+            },
+            Err(err) => panic!("{}", err), // If no orgs are found something went seriously wrong, no reason to go any further
+        }
+
+        println!("Finished SnykProvider scan, elapsed time in milis: ({:?})", start.elapsed().as_millis());
+
+        return valid_sboms;
+    }
+
+}
+
+pub struct SnykProvider{
+    snyk_api: Box<dyn Send + Sync + SnykRestClient>,
+}
+
+#[async_trait]
+impl SbomProvider for SnykProvider {
+
+    async fn provide_sboms(&self) {
+        // Get snyk access token
+        let snyk_token = get_snyk_access_token().await;
+
+        // Get and return all valid sboms found in Snyk
+        let valid_sboms = self.get_valid_sboms(snyk_token).await;
+
+        for sbom_results in valid_sboms.iter() {
+            //println!("Uploading SBOM to harbor for project: {}, from org: {}", sbom_results.1.id, orgs.org.id.clone().unwrap());
+            println!("Uploading SBOM to harbor for project: {}", sbom_results.1.id);
+            //TODO: Re-enable the next line
+            //simple_upload_sbom(cloud_front_domain.clone(), sbom_token.clone(), team_id.clone(), sbom_results.1.browse_url.clone(), sbom_results.1.r#type.clone(), sbom_results.0).await;
```

Comment:
Did you intend for this to be commented out?  Does it actually upload the SBOMs if you post?